frontend/projects/upgrade/src/app/features/dashboard/feature-flags/modals/enable-feature-flag-modal/enable-feature-flag-modal.component.ts

```diff
@@ -0,0 +1,59 @@
+import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
+import { CommonModalComponent } from '../../../../../shared-standalone-component-lib/components';
+import { MAT_DIALOG_DATA, MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { CommonModalConfig } from '../../../../../shared-standalone-component-lib/components/common-modal/common-modal-config';
+import { TranslateModule } from '@ngx-translate/core';
+import { FeatureFlagsService } from '../../../../../core/feature-flags/feature-flags.service';
+import { UpdateFeatureFlagStatusRequest } from '../../../../../core/feature-flags/store/feature-flags.model';
+import { FEATURE_FLAG_STATUS } from '../../../../../../../../../../types/src';
+import { Subscription } from 'rxjs';
+import { CommonModule } from '@angular/common';
+
+@Component({
+  selector: 'app-enable-feature-flag-modal-content',
+  standalone: true,
+  imports: [CommonModalComponent, TranslateModule, CommonModule],
+  templateUrl: './enable-feature-flag-modal.component.html',
+  styleUrl: './enable-feature-flag-modal.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class EnableFeatureFlagModalComponent {
+  subscriptions = new Subscription();
+  selectedFlag$ = this.featureFlagService.selectedFeatureFlag$;
+  isLoadingUpdateFeatureFlagStatus$ = this.featureFlagService.isLoadingUpdateFeatureFlagStatus$;
+  selectedFlagStatusChange$ = this.featureFlagService.selectedFeatureFlagStatusChange$;
+
+  get FEATURE_FLAG_STATUS() {
+    return FEATURE_FLAG_STATUS;
+  }
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA)
+    public config: CommonModalConfig,
+    public featureFlagService: FeatureFlagsService,
+    public dialog: MatDialog,
+    public dialogRef: MatDialogRef<EnableFeatureFlagModalComponent>
+  ) {}
+
+  ngOnInit() {
+    this.listenForFeatureFlagStatusChanges();
+  }
+
+  listenForFeatureFlagStatusChanges(): void {
+    this.subscriptions = this.selectedFlagStatusChange$.subscribe(() => {
+      this.closeModal();
+    });
```

Comment:
why add a subscriber to close the modal, you can add a close modal at the end of `primaryButtonClick` function. Are we waiting for a status change only after that do you want to change status? It doesn't seem necessary, there won't be much delay there.

Reply:
it is a requirement that we handle all loading states. a front-end developer cannot say "it's probably going to be quick". The pattern for user-initiated http requests is that we don't move on until the response is received.